internal/alb/tg/targethealth.go

```diff
@@ -0,0 +1,339 @@
+	} else {
+		// update condition
+		if updateTimes {
+			cond.LastProbeTime = now
+			if cond.Status != conditionStatus {
+				cond.LastTransitionTime = now
+			}
+		}
+		cond.Status = conditionStatus
+		cond.Reason = aws.StringValue(targetHealth.Reason)
+		cond.Message = aws.StringValue(targetHealth.Description)
+		pod.Status.Conditions[i] = *cond
+	}
+
+	// pod will always be updated (at least to update `LastProbeTime`);
+	// this will trigger another invocation of `Reconcile`, which will remove this pod from the list of pods to reconcile if its health status is ok
+	err := c.client.Status().Update(ctx, pod)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// From the given targets, only returns the ones that have a readiness gate for the given ingress/service and whose pod conditions actually need to be reconciled.
+func (c *targetHealthController) filterTargetsNeedingReconciliation(conditionType api.PodConditionType, t *Targets, desiredTargets []*elbv2.TargetDescription) ([]*elbv2.TargetDescription, error) {
+	targetsToReconcile := []*elbv2.TargetDescription{}
+	if len(desiredTargets) == 0 {
+		return targetsToReconcile, nil
+	}
+
+	// find the pods that correspond to the targets
+	pods, err := c.endpointResolver.ReverseResolve(t.Ingress, t.Backend, desiredTargets)
+	if err != nil {
+		return targetsToReconcile, err
+	}
+
+	// filter out targets whose pods don't have the `readinessGate` for this target group or whose pod condition status is already `True`
+	for i, target := range desiredTargets {
+		pod := pods[i]
+		if pod != nil && podHasReadinessGate(pod, conditionType) {
+			if _, cond := podConditionForReadinessGate(pod, conditionType); cond == nil || cond.Status != api.ConditionTrue {
+				targetsToReconcile = append(targetsToReconcile, target)
+			}
+		}
+	}
+
+	return targetsToReconcile, nil
+}
+
+func (c *targetHealthController) ingressTargetHealthReconciliationInterval(initial bool, serviceName string, ingress *extensions.Ingress) int64 {
+	ingressAnnos, err := c.store.GetIngressAnnotations(k8s.MetaNamespaceKey(ingress))
+	if err == nil {
+		serviceKey := types.NamespacedName{Namespace: ingress.Namespace, Name: serviceName}
+		serviceAnnos, err := c.store.GetServiceAnnotations(serviceKey.String(), ingressAnnos)
+		if err == nil {
+			if initial {
+				return *serviceAnnos.HealthCheck.IntervalSeconds * *serviceAnnos.TargetGroup.HealthyThresholdCount
+			}
+			return *serviceAnnos.HealthCheck.IntervalSeconds
+		}
+	}
+
+	if initial {
+		return healthcheck.DefaultIntervalSeconds * targetgroup.DefaultHealthyThresholdCount
+	}
+	return healthcheck.DefaultIntervalSeconds
+}
+
+// PodConditionTypeForIngressBackend returns the PodConditionType that is associated with the given ingress and backend
+func podConditionTypeForIngressBackend(ingress *extensions.Ingress, backend *extensions.IngressBackend) api.PodConditionType {
+	return api.PodConditionType(fmt.Sprintf(
+		"target-health.%s/%s_%s_%s",
+		parser.AnnotationsPrefix,
```

Comment:
i'd prefer to change this to hard-coded `alb.ingress.kubernetes.io` or better `alb.ingress.k8s.aws` (aligns with v2).

the parser.AnnotationsPrefix is dedicated for annotations, and people can override it with commandline arguments.

Reply:
So `target-health.alb.ingress.k8s.aws`? Is the `alb.ingress.k8s.aws` part already decided upon or might this change in the future?

Reply:
yes, we plan to use `alb.ingress.k8s.aws` for ingress related. more specifically, we plan to use `k8s.aws` as domain suffix for our CRDs as well.